app.rb

```diff
@@ -0,0 +1,134 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_books_with_index
+    @books.each_with_index { |book, i| puts "#{i}) Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people_with_index
+    @people.each_with_index do |person, i|
+      puts "#{i}) [#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    if student_or_teacher == 1
+      create_student
+    elsif student_or_teacher == 2
+      create_teacher
```

Comment:
- [ ] The app crashes when I try to create a teacher. Kindly fix that.

  ![image](https://github.com/GhostEsso/School_library/assets/81468675/4d8b711a-2776-47a0-a75f-61aa9ce30244)